pandas/core/indexes/base.py

```diff
@@ -3190,17 +3190,16 @@ def is_int(v):
         # convert the slice to an indexer here
 
         # if we are mixed and have integers
-        try:
-            if is_positional and self.is_mixed():
+        if is_positional and self.is_mixed():
+            try:
                 # Validate start & stop
                 if start is not None:
                     self.get_loc(start)
                 if stop is not None:
                     self.get_loc(stop)
                 is_positional = False
-        except KeyError:
-            if self.inferred_type in ["mixed-integer-float", "integer-na"]:
```

Comment:
this was not needed?

Reply:
not reachable; the check for self.is_mixed() a few lines up means these inferred_types are impossible